gammapy/scripts/cta_sensitivity.py

```diff
@@ -0,0 +1,250 @@
+            num = int((stop - start)/0.1)
+            fine_excess = np.linspace(start=start, stop=stop, num=num)
+            fine_on = fine_excess + bkg_counts[icount]
+            fine_off = np.zeros(len(fine_on)) + bkg_counts[icount] / self.alpha
+            fine_sigma = significance_on_off(n_on=fine_on, n_off=fine_off, alpha=self.alpha, method='lima')
+            idx = np.abs(fine_sigma - self.sigma).argmin()
+            if fine_excess[idx] >= self.gamma_min and fine_excess[idx] >= self.bkg_sys * bkg_counts[icount]:
+                excess[icount] = fine_excess[idx]
+            else:
+                excess[icount] = max(self.gamma_min, self.bkg_sys * bkg_counts[icount])
+
+            if self.verbosity > 0 :
+                print('N_ex={}, N_fineEx={}, N_bkg={}, N_bkgsys={}, Sigma={}'.format(excess[icount],fine_excess[idx], \
+                    bkg_counts[icount],self.bkg_sys * bkg_counts[icount],fine_sigma[idx]))
+
+        return excess
+
+    # Return differential fluxes
+    def get_1TeV_differential_flux(self, excess_counts, sp_model, aeff, edisp):
+        # Compute expected excess
+        predictor = CountsPredictor(sp_model, aeff=aeff, edisp=edisp, livetime=self.livetime)
+        predictor.run()
+        counts = predictor.npred.data.data.value
+        # Conversion in flux
+        flux = excess_counts / counts * u.Unit('1 / (cm2 TeV s)')
+
+        return flux
+
+    def run(self):
+
+        # Creation of the spectal shape
+        norm = 1 * u.Unit('1 / (cm2 s TeV)')
+        index = self.slope
+        ref = 1 * u.TeV
+        model = PowerLaw(index=index, amplitude=norm, reference=ref)
+
+        # Get the bins in reconstructed  energy
+        reco_energy = self.cta_perf.bkg.energy
+
+        # Start the computation
+        bkg_counts = self.get_bkg(self.cta_perf.bkg)
+        excess_counts = self.get_excess(bkg_counts)
+
+        Phi0 = self.get_1TeV_differential_flux(excess_counts, model, self.cta_perf.aeff, self.cta_perf.rmf)
+        diff_flux = (Phi0 * model.evaluate(energy=reco_energy.log_center(), index=index, amplitude=1, reference=ref) * \
+                     reco_energy.log_center() ** 2 * 1.60218).to('erg / (cm2 s)')
+
+        self.energy = reco_energy.log_center()
+        self.diff_sens = diff_flux
+
+    # Plot the result
+    def plot(self):
+
+        ax = plt.gca()
+        ax.plot(self.energy.value, self.diff_sens.value, color='red', label='GammaPy')
+        ax.set_xscale('log')
+        ax.set_yscale('log')
+        ax.grid(True)
+        ax.set_xlabel('Reco Energy [{}]'.format(self.energy.unit))
+        ax.set_ylabel('Sensitivity [{}]'.format(self.diff_sens.unit))
+        if self.verbosity > 0:
+            self.cta_perf.sens.plot(color='black', label="ROOT")
+            plt.legend()
+        # ax.errorbar(energy.value, values.value, xerr=xerr, fmt='o', **kwargs)
+
+        print("\n #################################\n\033[34;1m   Emid        Sensi[{}]\033[0m".format(self.diff_sens.unit))
```

Comment:
From here on out you don't `plot` any more, you `print` stuff.
This shouldn't be in the `plot` method, please move it to a separate method or property.

In this case, if I see correctly, you're just trying to pretty-print the result as a table.
It would be better if you expose a property or add a method that fills that data into an `astropy.table.Table` object and returns that.
A table will automatically pretty-print on the console and notebook.
And with one line the user can e.g. save it to a file.
So providing the results of the computation as a `Table` is better than custom table-printout code, I think that can be removed.